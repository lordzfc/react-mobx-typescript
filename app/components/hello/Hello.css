:local(.hello) {
  color: #F0F;
}