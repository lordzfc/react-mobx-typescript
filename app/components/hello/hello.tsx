import * as React from "react";
import * as styles from './Hello.css';


export interface HelloProps { compiler: string; framework: string; }

export class Hello extends React.Component<HelloProps, {}> {
    add(a:number , b: number):number {
        return a + b;
    }
    render() {
        return <h1 className={styles.hello}>Hello from {this.props.compiler} and {this.props.framework}!</h1>;
    }
}