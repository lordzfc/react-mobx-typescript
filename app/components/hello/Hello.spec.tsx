import * as React from 'react';
import * as TestUtils from 'react-dom/test-utils';
import ReactShallowRenderer = require('react-test-renderer/shallow');

import {Hello} from './hello'

describe('<Hello />', ()=>{
  const shallowRenderer = new ReactShallowRenderer();
  it('renders Hello', ()=>{
    expect(shallowRenderer.render(<Hello compiler="TypeScript" framework="React"/>))
    .toMatchSnapshot();
  })
  
})