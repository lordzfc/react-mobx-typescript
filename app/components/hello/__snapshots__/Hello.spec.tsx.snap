// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<h1
  className={undefined}
>
  Hello from 
  TypeScript
   and 
  React
  !
</h1>
`;

exports[`<Hello /> renders Hello 1`] = `
<h1
  className={undefined}
>
  Hello from 
  TypeScript
   and 
  React
  !
</h1>
`;
